discovery-handlers/onvif/src/discovery_impl.rs

```diff
@@ -195,156 +196,170 @@ pub mod util {
             .probe_matches
             .probe_match
             .iter()
+            .filter(|probe_match| !execute_filter(scopes, &probe_match.scopes))
             .flat_map(|probe_match| probe_match.xaddrs.split_whitespace())
             .map(|addr| addr.to_string())
             .collect::<Vec<String>>()
     }
 
+    // Accepted uris are both responsive and contain appropriate scopes
+    async fn get_qualifying_uris_from_discovery_response(
+        discovery_response: &str,
+        scopes: Option<&FilterList>,
+        onvif_query: &impl OnvifQuery,
+    ) -> Vec<String> {
+        let filtered_uris =
+            get_scope_filtered_uris_from_discovery_response(discovery_response, scopes);
+        let mut responsive_uris = Vec::new();
+        for uri in filtered_uris {
+            match onvif_query.is_device_responding(&uri).await {
```

Comment:
maybe for the next iteration, i wonder if we couldn't put each of these is_device_responding into a thread and then wait for all the threads to finish.  it seems like if there were 10 cameras and each wasn't responding, we'd have a 10 second loop here (sequentially waiting 1 second for each to timeout).  maybe putting each in its own thread would allow us to wait closer to 1 second total for all the queries?

Reply:
Thats a great point. I made changes to `simple_onvif_discover` to first get all filtered uris and then join all the futures that make calls to devices: c55d42feebfea32